Classes/Integration/HookSubscribers/DataHandlerSubscriber.php

```diff
@@ -124,19 +124,25 @@ public function processDatamap_preProcessFieldArray(array &$fieldArray, $table,
         // field. Updated value will still be subject to permission checks.
         $resolver = GeneralUtility::makeInstance(ObjectManager::class)->get(ProviderResolver::class);
         foreach ($fieldArray as $fieldName => $fieldValue) {
-            if ($GLOBALS["TCA"][$table]["columns"][$fieldName]["config"]["type"] === 'flex') {
+            if (isset($GLOBALS["TCA"][$table]["columns"][$fieldName]["config"]["type"]) && $GLOBALS["TCA"][$table]["columns"][$fieldName]["config"]["type"] === 'flex') {
                 $primaryConfigurationProvider = $resolver->resolvePrimaryConfigurationProvider(
                     $table,
                     $fieldName
                 );
 
                 if ($primaryConfigurationProvider && is_array($fieldArray[$fieldName]) && array_key_exists('data', $fieldArray[$fieldName])) {
                     foreach ($fieldArray[$fieldName]['data'] as $sheet) {
-                        foreach ($sheet['lDEF'] as $key => $value) {
-                            list ($possibleTableName, $columnName) = explode('.', $key, 2);
-                            if ($possibleTableName === $table && isset($GLOBALS['TCA'][$table]['columns'][$columnName])) {
-                                $fieldArray[$columnName] = $value['vDEF'];
-                            }
+                        foreach ($sheet['lDEF'] as $key => $value) 
```

Comment:
This chunk would need a whitespace/indentation and CGL fix for spaces and curly braces.

Reply:
Fix for spaces and curly braces